be/src/exec/es_http_scan_node.h

```diff
@@ -0,0 +1,112 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef  BE_EXEC_ES_HTTP_SCAN_NODE_H
+#define  BE_EXEC_ES_HTTP_SCAN_NODE_H
+
+#include <atomic>
+#include <condition_variable>
+#include <map>
+#include <string>
+#include <vector>
+#include <mutex>
+#include <thread>
+#include <future>
+
+#include "common/status.h"
+#include "exec/scan_node.h"
+#include "exec/es_http_scanner.h"
+#include "gen_cpp/PaloInternalService_types.h"
+
+namespace doris {
+
+class RuntimeState;
+class PartRangeKey;
+class PartitionInfo;
+class EsHttpScanCounter;
+class EsPredicate;
+
+class EsHttpScanNode : public ScanNode {
+public:
```

Comment:
Why exists two kinds of es scan node, only because the method access ES is different ?   ESScanNode only support single thread scanning but EsHttpScanNode support multi thread scanning, why?

Reply:
Due to history cause, It will be merged together in future